llvm/lib/Target/AIE/AIEInterBlockScheduling.cpp

```diff
@@ -169,7 +173,8 @@ bool InterBlockScheduling::leaveBlock() {
     // If we are very unlucky, we may step both the latency margin and
     // the resource margin to the max. Any more indicates failure to converge,
     // and we abort to prevent an infinite loop.
-    if (BS.FixPoint.NumIters > 2 * HR->getConflictHorizon()) {
+    if (BS.FixPoint.NumIters >
```

Comment:
Considering your change, does this error become more common without this increase?

Reply:
Oh it never triggered, I just changed the condition to account for the extra iterations, otherwise we would fail thinking we are in an infinite loop.